platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/dependency_locking.adoc

```diff
@@ -0,0 +1,301 @@
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingClasspathConfiguration/kotlin",files="build.gradle.kts[tags=locking-classpath]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingClasspathConfiguration/groovy",files="build.gradle[tags=locking-classpath]"]
+====
+
+== Generating and updating dependency locks
+
+To generate or update the lock state, use the `--write-locks` command line argument along with the tasks that would trigger configurations to be resolved:
+
+[source,text]
+----
+$ ./gradlew dependencies --write-locks
+----
+
+This will create or update the lock state for each resolved configuration during that build execution.
+If a lock state already exists, it will be overwritten.
+
+[source,text]
+.gradle.lockfile
+----
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.code.findbugs:jsr305:3.0.2=classpath
+com.google.errorprone:error_prone_annotations:2.3.2=classpath
+com.google.gradle:osdetector-gradle-plugin:1.7.1=classpath
+com.google.guava:failureaccess:1.0.1=classpath
+com.google.guava:guava:28.1-jre=classpath
+com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava=classpath
+com.google.j2objc:j2objc-annotations:1.3=classpath
+empty=
+----
+
+NOTE: Gradle won't write the lock state to disk if the build fails, preventing the persistence of potentially invalid states.
+
+[[lock_all_configurations_in_one_build_execution]]
+=== Lock all configurations in a single build execution
+
+When working with multiple configurations, you may want to lock them all at once in a single build execution.
+You have two options for this:
+
+1. **Run `gradle dependencies --write-locks`:**
+- This command will lock all resolvable configurations that have locking enabled.
+- In a multi-project setup, note that `dependencies` is executed only on one project, typically the root project.
+
+2. **Declare a Custom Task to Resolve All Configurations:**
+- This approach is particularly useful if you need more control over which configurations are locked.
+- It is not applicable to Android projects.
+
+This custom task resolves all configurations, locking them in the process:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/kotlin",files="build.gradle.kts[tags=resolve-all]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/groovy",files="build.gradle[tags=resolve-all]"]
+====
+
+This is especially useful in environments like native builds, where not all configurations can be resolved on a single platform.
+By filtering and resolving specific configurations, you ensure that only the relevant ones are locked, tailoring the locking process to your project's needs.
```

Comment:
> This is especially useful in environments like native builds

💅 It's the _filtering_ that is more than useful, it's _necessary_ to write multiple tasks, which run on each platform as required, when not all deps can be resolved via a single task.  The wording here makes it sound like _this example task_ is useful, which I think is misleading.

Reply:
I inverted the sentences to make it more clear.